server/routes/slack/slack_events.py

```
@@ -556,4 +589,207 @@ def _handle_suggestion_details(payload: dict, action: dict, slack_user_id: str,
+            dismissed.get("vcs_provider") or "github",
+            dismissed.get("repo_full_name") or "",
+            pr_number,
+            timeout=_GITHUB_CLOSE_TIMEOUT,
+        ) if pr_number else {"error": "No PR recorded for this recommendation"}
+
+        if close_result.get("already_merged"):
+            # A merge is acceptance, not rejection: it must not start an
+            # anti-nag window, or the next genuine drift goes unreported.
+            try:
+                with db_pool.get_admin_connection() as conn:
+                    with conn.cursor() as cursor:
```

Ruff (0.16.0)

[warning] Use a single `with` statement with multiple contexts instead of nested `with` statements

(SIM117)